script.js

```diff
@@ -0,0 +1,206 @@
+function addNewTodoLi(todo) {
+  // li에 필요한 element들 선언
+  const newLi = document.createElement('li');
+  const contentDiv = document.createElement('div');
+  const fromDateDiv = document.createElement('div');
+  const betweenDiv = document.createElement('div');
+  const toDateDiv = document.createElement('div');
+  const doneBtn = document.createElement('button');
+  const deleteBtn = document.createElement('button');
+
+  // elements의 내용 추가
+  contentDiv.innerHTML = todo.content;
+  contentDiv.className = 'content';
+  fromDateDiv.innerHTML = convertDate(todo.fromDate);
+  fromDateDiv.className = 'fromDate';
+  betweenDiv.innerHTML = '-';
+  betweenDiv.className = 'betweenDate';
+  toDateDiv.innerHTML = convertDate(todo.toDate);
+  toDateDiv.className = 'toDate';
+  doneBtn.innerHTML = todo.isDone ? '↪' : '✓';
+  doneBtn.className = 'doneBtn';
+  deleteBtn.innerHTML = '-';
+  deleteBtn.className = 'deleteBtn';
+
+  // event listener 추가
+  doneBtn.addEventListener('click', handleClickDoneBtn);
+  deleteBtn.addEventListener('click', handleClickDeleteBtn);
+
+  // li에 child node들 추가
+  newLi.appendChild(contentDiv);
+  newLi.appendChild(fromDateDiv);
+  newLi.appendChild(betweenDiv);
+  newLi.appendChild(toDateDiv);
+  newLi.appendChild(doneBtn);
+  newLi.appendChild(deleteBtn);
+
+  newLi.id = todo.idx; // 정렬을 위한 id 추가
+  // 각종 className 추가
+  newLi.classList.add('todoLi'); 
+  if (todo.priority === 3) newLi.classList.add('high');
+  else if (todo.priority === 2) newLi.classList.add('mid');
+  else newLi.classList.add('low');
+
+  if (todo.isDone) newLi.classList.add('done');
+  ul.appendChild(newLi);
+}
+
+// 기준에 따라 정렬되어 memory에 저장되도록 push, push할 때만 정렬하면 따로 정렬과정 거치지 않아도 됨
+function pushTodo(todo) {
+  if (todo.isDone) { // 완료한 todo는 맨 뒤에 위치
+    TODOS.splice(TODOS.length, 0, todo);
+    return;
+  }
+  let i = 0;
+  for (; i < TODOS.length; i++) {
+    if (TODOS[i].isDone) break; // 완료한 todo보다 무조건 앞에 위치
+    if (TODOS[i].priority < todo.priority) break; // priority가 더 높으면 그 앞에 위치
+    else if (TODOS[i].priority === todo.priority) { // priority가 같다면
+      if (TODOS[i].fromDate >= todo.fromDate) break; // 시작일이 빠른 todo가 더 낲에 위치
+    }
+  }
+  TODOS.splice(i, 0, todo);
+}
+
+// Date를 적절하게 display 하도록 convert
+function convertDate(date) {
+  return `${date.getFullYear()}.${date.getMonth() + 1}.${date.getDate()}`;
+}
+
+// ############################## click handlers ##############################
+// done button click event handler
+function handleClickDoneBtn(e) {
+  const idx = e.currentTarget.parentNode.id;
+  const clickedLi = document.getElementById(idx);
+  // class 추가/제거
+  if (clickedLi.classList.contains('done')) clickedLi.classList.remove('done');
+  else clickedLi.classList.add('done');
+
+  // TODOS에 저장된 todo 객체 수정
+  let i;
+  for (i = 0; i < TODOS.length; i++) {
+    if (TODOS[i].idx == idx) {
+      TODOS[i].isDone = !TODOS[i].isDone;
+      break;
+    }
+  }
+  const todo = TODOS[i];
+
+  TODOS.splice(i, 1); // 해당 todo를 삭제한 후
```

Comment:
이 부분도 고차함수를 이용할 수 있을 것 같아요!
```suggestion
  const todoIndex = TODOS.findIndex((todo) => todo.idx == idx);
  if (todoIndex !== -1) {
    const todo = TODOS[todoIndex];
    todo.isDone = !todo.isDone;
    TODOS.splice(todoIndex, 1);
}

```